src/lib.rs

```diff
@@ -44,13 +39,102 @@ pub fn hexf32(input: TokenStream) -> TokenStream {
 /// ```
 #[proc_macro]
 pub fn hexf64(input: TokenStream) -> TokenStream {
-    let lit = syn::parse_macro_input!(input as syn::LitStr);
-    match hexf_parse::parse_hexf64(&lit.value(), true) {
-        Ok(v) => format!("{:?}f64", v) // should keep the sign even for -0.0
-            .parse()
-            .expect("formatted a f64 literal"),
-        Err(e) => syn::Error::new(lit.span(), format!("hexf64! failed: {}", e))
-            .to_compile_error()
-            .into(),
+    parse_stream(input, |s, span| match hexf_parse::parse_hexf64(s, true) {
+        Ok(v) => TokenTree::Literal(Literal::f64_suffixed(v)).into(),
+        Err(e) => compile_error_stream(&format!("hexf64! failed: {}", e), span),
+    })
+}
+
```

Comment:
I would prefer to split the rest of this patch into a separate file, since a nitty-gritty of dealing with `TokenStream` shouldn't be the main concern here.

(I even feel that this parsing code has a bigger utility for other libraries and can be made into a small individual crate! If you feel like so and update this PR accordingly I will accept that too.)